CLI and lang updates assoc with tagging (#35668)
hashicorp/terraform: website/docs/cli/cloud/settings.mdx

## Define connection settings

Add a `cloud` block to your Terraform configuration and configure the connection settings to link the working directory to an HCP Terraform workspace. The `cloud` block is a member of the `terraform` block. Refer to the [`terraform` block reference](/terraform/language/terraform) for additional information.

Specify the following settings in the `cloud` block:

- `organization`:  Specifies the name of an HCP Terraform organization to connect to.
- `workspaces.tags`: Specifies a list of single-value string tags. Terraform links the working directory to existing workspaces in the organization that have matching tags. If there are no existing workspaces with matching tags, the Terraform CLI prompts you to create a new workspace that inherits the tags you specify in this field when you initialize the configuration.
- `workspaces.name`: You can specify the name of an existing workspace to associate with the Terraform configuration instead of using tags. If you configure the `name`, you cannot use the `tags` configuration.

> - `workspaces.project`: You can specify the name of an existing project to limit associating the configuration with workspaces in the specific project.

Author: This could be a little clearer. The way it was explained to me is that specifying the project associates the configuration with the all workspaces in the project that have the tags. It's a filtering mechanism so that you don't have to figure out which specific workspaces to associate the configuration with. I don't know how realistic this example is, but if you have a project called 'test' and maybe a set of workspaces tagged with `api`, then you could direct Terraform to associate the configuration with just those workspaces. 

We'll need to get some clarity from someone on what happens if the project does not exist, but Terraform creates a workspace using either the `name` or the `tags` value if they do not exist in the specified project. 

Some of this is already explained in the reference. If you are following this instructions, you might not have some of these questions because you already know that there is a project, workspace, or set of workspaces in HCP that you want to connect to. To answer those questions, we expect someone to follow the link to the complete reference.

> - `workspaces.project`: You can specify the name of an existing project to limit associating the configuration with workspaces in the specific project.

Author: ```suggestion
- `workspaces.project`: You can specify the name of an existing project. Terraform associates the configuration with workspaces in the project that match the `name` or `tags`. 
```

### Migrate remote backend

In the `terraform` block or `terraform.tf` file, replace `backend "remote"` with `cloud`. Terraform will continue to use the same ste of HCP Terraform workspaces.

The following example migrates the state data for a single workspace:

```hcl
terraform {
-  backend "remote" {
+    cloud {
       organization = "my-org"

       workspaces {
          name = "my-app-prod"
       }
     }
   }
}
```

If the `terraform` block or `terraform.tf` file uses the `prefix` argument to connect to multiple workspaces, you can specify a list of single-value string tags in the `tags` argument instead. During `terraform plan` or `terraform apply` operations, Terraform associates the configuration with workspaces that match the specified tags.

> The following example replaces the `my-app-` prefix with the `app:mine` tag:

Author: The code blocks have `-` and `+` to indicate how to change the file. I take your point, but I think because 1) we're making a concerted effort to stop referring to previous examples moving forward and 2) that when you see the example block that it will be clear. We can revisit if people struggle to understand.
